File: elza-core/src/main/java/cz/tacr/elza/controller/vo/nodes/ItemTypeLiteVO.java
package cz.tacr.elza.controller.vo.nodes;

import java.util.List;


/**
 * VO Odlehčená verze specifikace hodnoty atributu.
 *
 * @author Martin Šlapa
 * @since 11.2.2016
 */
public class ItemTypeLiteVO {

    /**
     * identifikator typu
     */
    private Integer id;

    /**
     * typ
     */
    private Integer type;

    /**
     * opakovatelnost
     */
    private Integer rep;

    /**
     * počítaný
     */
    private Integer cal;

    /**
     * stav počítanýho atributu
     */
    private Integer calSt;

    /**
     * seznam specifikací atributu
     */
    private List<DescItemSpecLiteVO> specs;

    /**
     * šířka atributu (0 - maximální počet sloupců, 1..N - počet sloupců)
     */
    private Integer width;

    public Integer getId() {
        return id;
    }

    public void setId(final Integer id) {
        this.id = id;
    }

    public Integer getType() {
        return type;
    }

    public void setType(final Integer type) {
        this.type = type;
    }

    public Integer getRep() {
        return rep;
    }

    public void setRep(final Integer rep) {
        this.rep = rep;
    }

    public List<DescItemSpecLiteVO> getSpecs() {
        return specs;
    }

    public void setSpecs(final List<DescItemSpecLiteVO> specs) {
        this.specs = specs;
    }

    public Integer getWidth() {
        return width;
    }

    public void setWidth(final Integer width) {
        this.width = width;
    }

    public Integer getCal() {
        return cal;
    }

    public Integer getCalSt() {
        return calSt;
    }

    public void setCalSt(final Integer calSt) {
        this.calSt = calSt;
    }

    public void setCal(final Integer cal) {
        this.cal = cal;
    }
}

File: elza/elza-react/src/api/ItemTypeLiteVO.ts
/**
 * VO Odlehčená verze specifikace hodnoty atributu.
 */
import {DescItemSpecLiteVO} from "./DescItemSpecLiteVO";

export interface ItemTypeLiteVO {
    /**
     * identifikator typu
     */
    id: number;

    /**
     * typ
     */
    type: number;

    /**
     * opakovatelnost
     */
    rep: number;

    /**
     * počítaný
     */
    cal: number;

    /**
     * stav počítanýho atributu
     * - 0 - vypnutý (použije se vypočtená hodnota)
     * - 1 - zapnutý (lze zadat hodnotu manuálně)
     *
     * Číslené z důvodu optimalizace
     */
    calSt: number;

    /**
     * atribut se může nastavit jako nedefinovaný
     * - 0 - nemůže
     * - 1 - může
     */
    ind: number;

    /**
     * seznam specifikací atributu
     */
    specs: DescItemSpecLiteVO[];

    /**
     * seznam identifikátorů oblíbených specifikací u typu
     */
    favoriteSpecIds: number[];

    /**
     * šířka atributu (0 - maximální počet sloupců, 1..N - počet sloupců)
     */
    width: number;
}
